### PR TITLE
Parity node support + a few small fixes

### PR DESCRIPTION
**Pull request by gyohng**

### CODE REVIEW

**Comment by gyohng** [comment]
Apologies for closing/reopening requests - was just doing some last minute testing to ensure that you get a high quality code.

File: egs/egs_ref.py [hunk 1]
```diff
@@ -117,18 +117,38 @@ def __init__(self):
...
+                    for tx_obj in tx_sequence.values():
+                        hashlist.append(tx_obj['hash'])
+                return hashlist
+            elif self.pending_method == 'parity':
+                txpoolpending = web3.manager.request_blocking('parity_pendingTransactions',[])
+                for tx_obj in txpoolpending:
+                    hashlist.append(tx_obj['hash'])
+                return hashlist
+        except ValueError as ve:
+            if ve.args[0]['code']==-32601: # method not found
```
Comment (10a7) [inline egs/egs_ref.py]: I'd prefer to see the expected node type abstracted to a command-line flag or a setting in the configuration file, vs. doing this eat-the-exception-and-guess-again method.

File: egs/jsonexporter.py [hunk 2]
```diff
@@ -74,10 +75,14 @@ def _serialize(self, mixed):
             mixed = json.loads(mixed)
         elif isinstance(mixed, dict):
             # web3 3.x -> 4.x: bytes() is not serializable
-            # first-degree HexBytes check as a final trap
+            # also pandas sometimes returns int64
```
Comment (10a7) [inline egs/jsonexporter.py]: 👍

**Comment by gyohng** [comment]
Parity versions tested: 1.10.x, 1.11.x, 1.12.x nightly
For it to work you need the 'parity' API to be enabled for JSON-RPC.

The guess-again logic: there's an argument for it, as I had a cluster of different nodes mapped through a fallback mechanism to the same port, so you'd never know which one you'd be getting. There's no downside, as it doesn't try it every time, but guesses just once, and then remembers the successful one.

**Comment by 10a7** [comment]
> The guess-again logic: there's an argument for it, as I had a cluster of different nodes mapped through a fallback mechanism to the same port, so you'd never know which one you'd be getting. There's no downside, as it doesn't try it every time, but guesses just once, and then remembers the successful one.

Yeah, I considered that as well - load balancing in a mixed Geth/Parity environment, you'd either have to have a LUT of some sort for what type of node you're talking to, or attempt to autodetect it like you're doing. I thought about this a bit and I'm OK with it.

However, there's another thing I forgot: if we're going to add this support we should probably also add something small to the `README.md` that explains that Parity support is also there, and how to enable it to use the mixed environment. Write up a little something into README and I'll test/merge the changeset. I don't have a Parity node to test against in my development environment right now.

**Comment by gyohng** [comment]
Please check. The summary of changes:
 - README.md updated, Parity is now there like if it always were
 - [geth] section renamed to [rpc] in the config, however it'll still read the old config files.

What's your public ssh key? I can set up a synced parity node with the root access for you and spin it for a week or so.

**Comment by 10a7** [comment]
@gyohng https://api.github.com/users/10a7/keys

**Comment by 10a7** [comment]
Blocked on:

1. [ethgasstation-api/#2](https://github.com/ethgasstation/ethgasstation-api/issues/2)
2. Parity testing in dev
3. Mixed geth/parity testing

@ethgasstation This project has picked up some steam. It's probably time for us to consider actually writing a test suite ;)

**Comment by gyohng** [comment]
It will read the old configs, check the code.

**Comment by ethgasstation** [comment]
Can you explain to me what this means?

**Comment by gyohng** [comment]
The latter bit is in the settings parsing code. If the [rpc] section is not explicitly present, it'll first check if [geth] is present and clone its contents into [rpc]. So the config files with [geth], [parity] or [rpc] will work. In other words, if you upgrade your infrastructure with the latest code, it won't break anything.

**Comment by gyohng** [comment]
@10a7 The IP address of the server is 159.69.32.70, you can ssh into it as root

egs is running under screen session already with the log written to /root/egslog.txt

Latest commits to my branch include the timeout parameter support. This node's hardware is not particularly fast (some cloud VM), and some RPC requests take up to 1 minute to complete. I figured, it'd be a useful generic parameter (as I saw EGS throwing timeout errors before), so I added it. 

Parity is running via systemd. To see parity log with the RPC debug, type journalctl -f -u parity

To see RPC completion times: journalctl -f -u parity | grep Took

Also, it looks like EGS doesn't handle timeouts gracefully (nothing due to my changes though) - you can reduce the timeout setting in the settings file to see what happens. It will pretty violently throw exceptions from all over the place.

**Comment by 10a7** [comment]
@gyohng I ended up getting a parity node of my own running, you can take this one down. Thanks.

**Comment by gyohng** [comment]
Thanks, it's taken down now.

**Pull request merged**